Packs/AzureKeyVault/Integrations/AzureKeyVault/AzureKeyVault.py

```diff
@@ -128,63 +133,76 @@ def create_or_update_key_vault_request(self, vault_name: str, object_id: str, lo
+        return self.http_request('PUT', full_url=full_url, data=data)
 
-        return response
-
-    def delete_key_vault_request(self, vault_name: str) -> Dict[str, Any]:
+    def delete_key_vault_request(self, subscription_id: str, resource_group_name: str,
+                                 vault_name: str) -> Dict[str, Any]:
         """
         Delete Key Vault by name.
 
         Args:
+            subscription_id (str): Subscription ID.
+            resource_group_name (str): Resource group name.
             vault_name (str): Key Vault name.
 
         Returns:
             Dict[str, Any]: API response from Azure.
         """
-        url_suffix = f'/vaults/{vault_name}'
-        response = self.http_request('DELETE', url_suffix=url_suffix)
+        full_url = f'https://management.azure.com/subscriptions/{subscription_id}/resourceGroups/' \
+            f'{resource_group_name}/providers/Microsoft.KeyVault/vaults/{vault_name}'
 
-        return response
+        return self.http_request('DELETE', full_url=full_url)
 
-    def get_key_vault_request(self, vault_name: str) -> Dict[str, Any]:
+    def get_key_vault_request(self, subscription_id: str, resource_group_name: str,
+                              vault_name: str) -> Dict[str, Any]:
         """
         Retrieve Key Vault by name.
 
         Args:
+            subscription_id (str): Subscription ID.
+            resource_group_name (str): Resource group name.
             vault_name (str): Key Vault name.
 
         Returns:
             Dict[str, Any]: API response from Azure.
         """
-        url_suffix = f'/vaults/{vault_name}'
-        response = self.http_request('GET', url_suffix=url_suffix)
-        return response
+        full_url = f'https://management.azure.com/subscriptions/{subscription_id}/resourceGroups/' \
+            f'{resource_group_name}/providers/Microsoft.KeyVault/vaults/{vault_name}'
+        return self.http_request('GET', full_url=full_url)
 
-    def list_key_vaults_request(self, limit: int, offset: int) -> List[dict]:
+    def list_key_vaults_request(self, subscription_id: str = None,
+                                limit: int = DEFAULT_LIMIT, offset: int = DEFAULT_OFFSET) -> List[dict]:
         """
         List Key Vaults by limit and offset arguments from the specified resource group and subscription.
 
         Args:
+            subscription_id (str): Subscription ID.
             limit(int): limit the number of key vaults to return.Default is 50.
             offset(int): First index to retrieve from. Default is 0.
         Returns:
             Dict[str, Any]: API response from Azure.
         """
-        url_suffix = '/vaults'
+        ful_url = f'https://management.azure.com/subscriptions/{subscription_id}/providers/Microsoft.KeyVault/' \
+            f'vaults?$top={limit}'
         response = self.http_request(
-            'GET', url_suffix=url_suffix)
+            'GET', full_url=ful_url)
+        if response.get('error'):
```

Comment:
Would be better to check the status code of the response - also - I think it should be handled in the self.http_request func please verify that.

Reply:
done.